middlewares/validators/user.js

```diff
@@ -13,22 +13,44 @@ const updateUser = async (req, res, next) => {
     .keys({
       phone: joi.string().optional(),
       email: joi.string().optional(),
-      username: joi.string().optional(),
+      username: joi
+        .string()
+        .optional()
+        .min(4)
+        .max(20)
+        .regex(/^(?=.*[a-zA-Z])(?=.*\d)[a-zA-Z0-9]+$/)
+        .message("Username must be between 4 and 20 characters long and contain only letters and numbers."),
```

Comment:
Thanks @mailmesriza98 for noticing the change, yes it should be changed to or because we are keeping both numbers/letters, This particular change is for sign-up i.e. for new users joining, it will not affect the users already present in the database.